server/gui_files/multiplayer.py

```diff
@@ -124,6 +124,23 @@ def record_wpm(name, user, wpm, token):
             db("DELETE FROM leaderboard WHERE user_id = (%s)", [user])
             db("INSERT INTO leaderboard (name, user_id, wpm) VALUES (%s, %s, %s)", [name, user, wpm])
 
+    @route
+    @forward_to_server
+    def check_on_leaderboard(user):
+        with connect_db() as db:
+            vals = db("SELECT wpm FROM leaderboard ORDER BY wpm DESC LIMIT 20").fetchall()
+            threshold = vals[-1][0] if len(vals) >= 20 else 0
+            ret = db("SELECT * FROM leaderboard WHERE user_id=(%s) AND wpm >= (%s)", [user, threshold]).fetchone()
```

Comment:
We can just turn this into one SQL query, no? Just check if the returned top 20 contains the user ID we’re looking for.

Reply:
True, that makes more sense.

Reply:
Change made